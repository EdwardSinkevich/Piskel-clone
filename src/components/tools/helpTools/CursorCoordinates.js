export default class CursorCoordinates {
  static showCoords(x, y) {
    const cursorCoordinatesContainer = document.getElementById('cursor-coordinates');
    cursorCoordinatesContainer.classList.add('visible');
    cursorCoordinatesContainer.innerHTML = `x=${x}, y=${y}`;
  }

  static hideCoords() {
    const cursorCoordinatesContainer = document.getElementById('cursor-coordinates');
    cursorCoordinatesContainer.classList.remove('visible');
  }

  render() {
    const animatedPreview = document.getElementById('animated-preview-container');
    const cursorCoordinatesContainer = document.createElement('div');
    cursorCoordinatesContainer.className = 'cursor-coordinates';
    cursorCoordinatesContainer.id = 'cursor-coordinates';

    animatedPreview.appendChild(cursorCoordinatesContainer);
  }
}
