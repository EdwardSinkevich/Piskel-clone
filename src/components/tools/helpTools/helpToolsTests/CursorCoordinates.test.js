import CursorCoordinates from '../CursorCoordinates';

describe('CursorCoordinates.showCoords(x, y)', () => {
  it('Should set coords x, y to cursor coordinates container innerHTML', () => {
    const cursorCoordinatesContainer = document.createElement('div');
    cursorCoordinatesContainer.id = 'cursor-coordinates';
    document.body.appendChild(cursorCoordinatesContainer);
    CursorCoordinates.showCoords(42, 42);
    expect(cursorCoordinatesContainer.innerHTML).toEqual('x=42, y=42');
  });
});

describe('CursorCoordinates.hideCoords', () => {
  it('Should remove class visible from cursor coordinates container', () => {
    CursorCoordinates.hideCoords();
    const cursorCoordinatesContainerAfter = document.getElementById('cursor-coordinates');
    expect(cursorCoordinatesContainerAfter.classList.contains('visible')).toBe(false);
  });
});

describe('CursorCoordinates.render', () => {
  it('Should render correct', () => {
    const animatedPreview = document.createElement('div');
    animatedPreview.id = 'animated-preview-container';
    document.body.appendChild(animatedPreview);
    const cursorCoordinates = new CursorCoordinates();
    cursorCoordinates.render();
    expect(document.getElementById('animated-preview-container').innerHTML).toMatchSnapshot();
  });
});
