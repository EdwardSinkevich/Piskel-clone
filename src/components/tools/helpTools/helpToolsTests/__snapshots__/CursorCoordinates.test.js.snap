// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CursorCoordinates.render Should render correct 1`] = `"<div class=\\"cursor-coordinates\\" id=\\"cursor-coordinates\\"></div>"`;
